ls1intum/Artemis
# [Feature] Grading System - Exam Summary - Grade Key Overview

### Checklist

### Motivation and Context


This is a follow-up to the previous student exam overview PR. The students can now view the grade key used for the exam on a separate page (and export it), after the results have been published

### Description

I created a new page with the grade key overview which can be accessed through the exam summary page, as long as the results of the exam have been published.

### Steps for Testing


1. Log in to Artemis as an instructor and create a grade key of type 'Grade' for an exam with student submissions (my course IvoTC on TS3 already has an exam with the appropriate submissions) 
2. Log in as a student that submitted in the said exam -> navigate to the exam summary page
3. A 'View Grade Key button should be present below the student's grade -> click on it
4. Check that the grade key is displayed correctly and the student's grade is highlighted (also test exporting)
5. Again log in as an instructor and change the Grade Key to type 'Bonus'
6. Go through steps 2-4 again and check if the appropriate changes exists for the new grade key
7. Finally check if the 'View Grade Key' button is not present if no grade key exists for the exam

### Test Coverage



grade-key-overview.component.ts: 96.55%
grading-system.service.ts: 100%
GradeStepResource.java: 100%

### Screenshots


![Screenshot 2021-05-31 085137](https://user-images.githubusercontent.com/63168603/120146314-910eac80-c1ed-11eb-8dcd-fad24f2451e0.png)
![Screenshot 2021-05-31 085156](https://user-images.githubusercontent.com/63168603/120146317-923fd980-c1ed-11eb-8bd9-839d12fa12ea.png)
![Screenshot 2021-05-31 085007](https://user-images.githubusercontent.com/63168603/120146318-923fd980-c1ed-11eb-9020-6cbcb50068d4.png)
